### DISCUSSION TITLE
BitECS next?

### DISCUSSION THREAD

**Discussion by Keristero** [NateTheGreatt/bitECS#139] [General]
Hi, I'm working on a tiny engine of sorts for my personal projects based on the current master branch, I was poking around to see if there is a way for me to deserialize replace (removing entities that don't exist in the serialized snapshot).

I see there is a very active branch, is there anywhere I can can read about the upcoming changes or inspiration for them?

**Comment by WestonVincze** [comment]
I'm in the same boat as Keristero. I'm curious about what the "next" branch is and I'd love to know more about it! I'm currently using the "main" branch and I noticed that "next" is far more active and 200 commits ahead of "main".

**Comment by krispya** [comment]
Hi everyone, the next branch is a rewrite of bitECS that adds a bunch of new features like prefabs, relationships, queues and ranges while maintaining (mostly) interoperability with bitECS v0.3.4. It will also perform better and suggest current day best practices like using native JS arrays instead of typed arrays. You can get a sneak peak by checking out the in progress docs [here](https://github.com/NateTheGreatt/bitECS/blob/next/packages/classic/README.md). 

There is no time table for release or even milestones for what we are looking to accomplish before releasing it. The next branch is very much a work in progress, but we will be publishing alphas time to time and move to a beta once things settle. More to come.

**Comment by dxu** [comment]
@krispya, is development of the next branch open to external contributors? If one was interested in getting involved, what would be the best way to do so?

**Reply by krispya** [reply]
Sure is. The source is on the next branch. You can join our Discord or just ask questions here. 

https://discord.gg/RU2kBdZE

**Reply by justynspooner** [reply]
> Sure is. The source is on the next branch. You can join our Discord or just ask questions here.
> 
> https://discord.gg/RU2kBdZE

I'm also interested in checking out the next branch and learning more. Have you got a non-expired link for Discord? Would love to join. Thanks.

**Reply by krispya** [reply]
New link: https://discord.gg/h28U8eyW

**Reply by bohdanzahurskyi1** [reply]
@krispya Hey, do you have a new link for Discord?

**Reply by vegeta897** [reply]
@bohdanz1997 https://discord.gg/aTGQQ8sAPy
